crowd_anki/representation/deck.py

```diff
@@ -137,7 +139,7 @@ def save_to_collection(self,
 
         if save_note_models:
             for note_model in self.metadata.models.values():
-                note_model.save_to_collection(collection)
+                note_model.save_to_collection(collection, import_config.use_note_models)
```

Comment:
um, aren't `save_note_model` and `import_config.use_note_models` identical in intent? if so shouldn't the latter replace the former?

Reply:
the metadata is shared between whole deck hierarchy. I'm still confused why these are different, let's chat when you're online

Reply:
So you suggested the following in our WhatsApp chat:

> I think it'd make sense to read the config when you call the function from outside the deck and pass in save_to_collection as read in by import_config.

Tbh I have no idea what you mean, even looking at the code. Gonna need some clarification there :sweat_smile: 

*However!* In an attempt to tidy up this code in my own way I have extrapolated out the saving children and notes (the recursive part) into a separate function, and gotten rid of most of the predefined variables littering it up. Take a look at the new system and let me know how you like it. I have tested a few imports with child decks and it works fine :+1:

Reply:
The new structure makes sense!